src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clones = [];
+
+  for (const action of actions) {
+    const currentState = clones[clones.length - 1] || state;
+    const clone = {};
+
+    if (action.type === 'addProperties') {
+      Object.assign(clone, currentState, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      for (const key in currentState) {
+        if (!action.keysToRemove.includes(key)) {
+          clone[key] = currentState[key];
+        }
+      }
+    }
```

Comment:
Good work but let's rewrite it to the switch/case statement since we rely on a single action.type value

Reply:
rewrote